contract/src/lib.rs

```diff
@@ -404,6 +409,69 @@ impl BtcLightClient {
         }
     }
 
+    fn check_target(&self, block_header: &Header, prev_block_header: &ExtendedHeader) {
+        if prev_block_header.block_height % BLOCKS_PER_ADJUSTMENT != 0 {
+            require!(
+                block_header.bits == prev_block_header.block_header.bits,
+                format!(
+                    "Error: Incorrect bits. Expected bits: {}; Actual bits: {}.",
+                    prev_block_header.block_header.bits, block_header.bits
+                )
+            );
+            return;
+        }
+
+        if prev_block_header.block_height + 1 >= BLOCKS_PER_ADJUSTMENT {
+            if let Some(init_header_hash) = self
+                .mainchain_height_to_header
+                .get(&(prev_block_header.block_height + 1 - BLOCKS_PER_ADJUSTMENT))
+            {
+                let init_extend_header = self
+                    .headers_pool
+                    .get(&init_header_hash)
+                    .unwrap_or_else(|| env::panic_str(ERR_KEY_NOT_EXIST));
+                let actual_time_taken =
+                    prev_block_header.block_header.time - init_extend_header.block_header.time;
+                let last_target = prev_block_header.block_header.target();
+
+                let (mut new_target, new_target_overflow) =
+                    last_target.overflowing_mul(u64::from(actual_time_taken));
+                require!(!new_target_overflow, "new target overflow");
+                new_target = new_target / U256::from(EXPECTED_TIME);
+
+                let (max_target, max_target_overflow) =
+                    last_target.overflowing_mul(MAX_ADJUSTMENT_FACTOR);
+                require!(!max_target_overflow, "max target overflow");
+
+                new_target = min(new_target, max_target)
```

Comment:
Do you think that it is better to adjust the `new_target` instead of `actual_time_taken` like it is done in bitcoin [core](https://github.com/bitcoin/bitcoin/blob/4835bba2cb1350259c20bf0faf852187203ba336/src/pow.cpp#L56)?

Reply:
fixed: https://github.com/Near-One/btc-light-client-contract/pull/64/commits/8a5de4d656352cee35d27719fbba840855ce03d4